src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -21,180 +21,180 @@ internal sealed class TaskCenterSolutionAnalysisProgressReporter
+
+        /// <summary>
+        /// Stores the last shown <see cref="ProgressData"/>
+        /// </summary>
+        private ProgressData _lastProgressData;
+
+        /// <summary>
+        /// Unfortunately, <see cref="ProgressData.PendingItemCount"/> is only reported
+        /// when the <see cref="ProgressData.Status"/> is <see cref="ProgressStatus.PendingItemCountUpdated"/>
+        /// So we have to store this in addition to <see cref="_lastProgressData"/> so that we
+        /// do not overwrite the last reported count with 0.
+        /// </summary>
+        private int _lastProgressCount;
 
         [ImportingConstructor]
         public TaskCenterSolutionAnalysisProgressReporter(
             SVsTaskStatusCenterService taskStatusCenterService,
             IDiagnosticService diagnosticService,
             VisualStudioWorkspace workspace)
         {
-            _lastTimeReported = DateTimeOffset.UtcNow;
-            _resettableDelay = null;
-
-            ResetProgressStatus();
-
             _taskCenterService = (IVsTaskStatusCenterService)taskStatusCenterService;
-
             _options = new TaskHandlerOptions()
             {
                 Title = ServicesVSResources.Running_low_priority_background_processes,
                 ActionsAfterCompletion = CompletionActions.None
             };
 
-            var crawlerService = workspace.Services.GetService<ISolutionCrawlerService>();
+            var crawlerService = workspace.Services.GetRequiredService<ISolutionCrawlerService>();
             var reporter = crawlerService.GetProgressReporter(workspace);
 
-            StartedOrStopped(reporter.InProgress);
+            if (reporter.InProgress)
+            {
+                Started();
+            }
+            else
+            {
+                Stopped();
+            }
 
-            // no event unsubscription since it will remain alive until VS shutdown
             reporter.ProgressChanged += OnSolutionCrawlerProgressChanged;
         }
 
-        private void OnSolutionCrawlerProgressChanged(object sender, ProgressData progressData)
+        /// <summary>
+        /// Retrieve and throttle solution crawler events to be sent to the progress reporter UI.
+        /// 
+        /// there is no concurrent call to this method since ISolutionCrawlerProgressReporter will serialize all
+        /// events to preserve event ordering
+        /// </summary>
+        /// <param name="progressData"></param>
+        public void OnSolutionCrawlerProgressChanged(object sender, ProgressData progressData)
+        {
+            lock (_lock)
+            {
+                _lastProgressData = progressData;
+
+                // The task is running which will update the progress.
+                if (_intervalTask != null)
+                {
+                    return;
+                }
+
+                // Report progress immediately to ensure we update the UI on the first event.
+                ReportProgress();
```

Comment:
The events could come in as Start -> some number of pause / evaluate.  Need to make sure I catch the start first so that I can create the task center task.

Reply:
also, this just seems busted.  if you are always calling this in response to OnSolutionCrawlerProgressChanged, then the delayed work is pointless.  this line needs to go afaict.